server/routes.js

```diff
@@ -172,29 +183,42 @@ app.route("/residents")
 			}, err => {
 				return Promise.resolve().then( () => { throw err; } )
 			})
-			// Inserts a new person into db if not found. Otherwise verifies that the person is not already a resident somewhere else.
+			// Inserts a new person into db if not found. Otherwise selects and verifies user is not currently residing in a house
 			.then(pRows => {
 				personResults = pRows;
 				if (personResults.length < 1) {
-					return con.query('INSERT INTO PEOPLE (FIRST_NAME, LAST_NAME, BIRTHDAY, ROLE_ID, VID) VALUES (?, ?, ?, ?, ?)', [newRes.fName, newRes.lName, newRes.birthday, 1, villageResults[0].VID])
+					return con.query('INSERT INTO PEOPLE (FIRST_NAME, LAST_NAME, BIRTHDAY, ROLE_ID, VID, GENDER, EMPLOYMENT, IDENTIFICATION, PREVIOUS_RESIDENCE, DISABILITIES, CHILDREN, PREVIOUS_SHELTER_PROGRAM, CRIMINAL_HISTORY) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)', 
+						[newRes.fName, newRes.lName, newRes.birthday, 1, villageResults[0].VID, newRes.gender, newRes.employment, newRes.identification, newRes.pastResidence, newRes.disabilities, newRes.children, newRes.pastShelter, newRes.criminalHistory])
 				} else {
 					return con.query('SELECT IN_RESIDENCE FROM RESIDENTS WHERE PID = ?', personResults[0].PID);
-					// return Promise.resolve(null);
 				}
 			}, err => {
 				return Promise.resolve().then( () => { throw err; } )
 			})
+			// Returns the inserted row if the previous promise executed an insert, otherwise verifies that the person is not already a resident somewhere else. Then updates existing people information.
```

Comment:
Can you explain `otherwise verifies that the person is not already a resident somewhere else`?

Reply:
Users should only have one residence at any given time. This check is required to prevent a person from being added as a resident to more than one house.

Reply:
I believe that there should also be a check that prevents two people to be assigned the same house.

Reply:
Does that imply we aren't keeping track of children that might be in the same house as the primary resident?

Reply:
We will need to ask LIHI is they allow this.
I'm more concerned about spouse than children